### PR TITLE
Backport fix for issue #14148 to release version 1.18.x

### PR DESCRIPTION
**Pull request by nickchapman-da**
Backport fix for issue #14148

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala [hunk 1]
```diff
@@ -200,6 +201,7 @@ private[lf] object PartialTransaction {
     keys = Map.empty,
     globalKeyInputs = Map.empty,
     localContracts = Set.empty,
+    localContractsEver = Set.empty,
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:204]: Why do we need to track both `localContracts` and `localContractsEver` at the level of PartialTransaction (I get why we need localContracts in ActiveLedgerState)? Before `localContracts` here corresponded to `localContractsEver` and afaict we never need the other one at this level

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:204]: Indeed I think we just need to rename localContracts to `localContractsEver`.

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:204]: Thanks @nickchapman-da. 
I just think the API in 1.18  is complelty broken, active state should be a field of `PartialTransaction`. 
I think you have made the good choice by not refactoring more, we want the change to be minimal to ease any future backport.

Comment (cocreature) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:204]: > The fields of the ActiveLedgerState are duplicated/flattenned in the PartialTransaction state (with no activeState indirection as we have in main)

This is true for `keys` and `consumedBy`. It is not true for `localContracts`. In 1.18, `localContracts` was not reset on a rollback so it corresponds to `localContractsEver` and there was nothing that tracked `locallyCreatedThisTimeline`.

Now I agree that this is pretty messy and we would be better off storing an `ActiveLedgerState` on the `PartialTransaction` but regardless of which approach we take, we should at least take this as an opportunityt to document these fields clearly.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala [hunk 2]
```diff
@@ -270,6 +270,12 @@ private[lf] object PartialTransaction {
   *
   *  @param actionNodeLocations The optional locations of create/exercise/fetch/lookup nodes in pre-order.
   *   Used by 'locationInfo()', called by 'finish()' and 'finishIncomplete()'
+  *
+  * @param localContractsEver
+  *   Tracks all contracts created by this transaction including those under a rollback.
+  *
+  * @param localContracts
+  *   Tracks contracts created by this transaction that have not been rolled back. This is a subset of `localContractsEver`.
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:278]: Thanks.

**Pull request merged**